File: backend/src/main/java/by/fpmibsu/findafriend/controller/commands/users/CreateUserHandler.java
package by.fpmibsu.findafriend.controller.commands.users;

import by.fpmibsu.findafriend.application.mediatr.RequestHandler;
import by.fpmibsu.findafriend.dataaccesslayer.user.UserDao;
import by.fpmibsu.findafriend.entity.Contacts;
import by.fpmibsu.findafriend.entity.User;
import by.fpmibsu.findafriend.services.PasswordHasher;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.ArrayList;

public class CreateUserHandler extends RequestHandler<Boolean, CreateUserCommand> {
    private static final Logger logger = LogManager.getLogger(CreateUserHandler.class);
    private final UserDao userDao;
    private final PasswordHasher passwordHasher;

    public CreateUserHandler(UserDao userDao, PasswordHasher passwordHasher) {
        this.userDao = userDao;
        this.passwordHasher = passwordHasher;
    }

    @Override
    public Boolean handle(CreateUserCommand request) {
        var contacts = new Contacts(request.name, request.surname, request.phoneNumber, request.email);
        var hashedPassword = passwordHasher.hashPassword(request.password);
        var user = new User(0, contacts, new ArrayList<>(), new ArrayList<>(), User.Role.USER, request.login, hashedPassword);
        try {
            userDao.create(user);
            logger.info(String.format("User %s just registered", request.login));
            return true;
        } catch (Exception e) {
            return false;
        }
    }
}

File: BackendNET/Backend/Commands/Users/CreateUser/CreateUserHandler.cs
using Backend.Entities;
using Backend.Repository;
using MediatR;
using Microsoft.AspNetCore.Identity;

namespace Backend.Commands.Users.CreateUser;

public class CreateUserHandler : IRequestHandler<CreateUserCommand, bool>
{
    private readonly IUserRepository _userRepository;
    private readonly IPasswordHasher<User> _passwordHasher;

    public CreateUserHandler(IUserRepository userRepository, IPasswordHasher<User> passwordHasher)
    {
        _userRepository = userRepository;
        _passwordHasher = passwordHasher;
    }

    public async Task<bool> Handle(CreateUserCommand request, CancellationToken cancellationToken)
    {
        var user = new User {
            Name = request.Name,
            Surname = request.Surname,
            Email = request.Email,
            PhoneNumber = request.PhoneNumber,
            Role = User.UserRole.User,
            Login = request.Login
        };

        var password = _passwordHasher.HashPassword(user, request.Password);
        user.Password = password;
        await _userRepository.InsertAsync(user, cancellationToken);
        return true;
    }
}